File: ASProj/app/src/main/java/org/devio/as/proj/main/HiApplication.java
package org.devio.as.proj.main;

import com.alibaba.android.arouter.launcher.ARouter;
import com.google.gson.Gson;

import org.devio.as.proj.ability.HiAbility;
import org.devio.as.proj.common.flutter.HiFlutterCacheManager;
import org.devio.as.proj.common.ui.component.HiBaseApplication;
import org.devio.hi.library.crash.CrashMgr;
import org.devio.hi.library.log.HiConsolePrinter;
import org.devio.hi.library.log.HiLogConfig;
import org.devio.hi.library.log.HiLogManager;
import org.devio.hi.library.util.ActivityManager;
import org.devio.hi.library.crash.CrashHandler;

import dagger.hilt.android.HiltAndroidApp;

@HiltAndroidApp
public class HiApplication extends HiBaseApplication {
    @Override
    public void onCreate() {
        super.onCreate();
        HiLogManager.init(new HiLogConfig() {
            @Override
            public JsonParser injectJsonParser() {
                return src -> new Gson().toJson(src);
            }
        }, new HiConsolePrinter());
        ActivityManager.getInstance().init(this);
        ARouter.init(this);
//        HiFlutterCacheManager.getInstance().preLoad(this);
        HiFlutterCacheManager.getInstance().preLoadDartVM(this);
//        CrashMgr.INSTANCE.init();
//        HiAbility.INSTANCE.init(this,"Umeng",null);
    }
}

File: hi_hilt/app/src/main/java/org/devio/as/proj/hi_hilt/HiApplication.kt
package org.devio.`as`.proj.hi_hilt

import android.app.Application
import dagger.hilt.android.HiltAndroidApp
/*1、编译时生成Hilt_HiApplication入口类，负责创建applicationComponet组件对象*/
/*2、编译时把父类从Application替换成Hilt_HiApplication*/
@HiltAndroidApp
class HiApplication: Application() {
    override fun onCreate() {
        super.onCreate()/*3、执行父类Hilt_HiApplication.onCreate()*/
        /*4、创建与Application生命周期关联的HiltComponents_HiApplicationC组件*/
        /*5、调用injectHiApplication为Application注入对象*/
    }
}